### PR TITLE
Percentile Group Customization for Mapbox

### PR DESCRIPTION
**Pull request by DanielManning**
Extends Pixiedust Mapbox visualizations to allow for more flexible customization of percentile group data.  This work completed by the IBM New Hire Developers cohort 2 Littleton group address two hills:

1. Data scientists using Pixiedust can cluster geospatial data points into percentile groups completely customized on data quantity and size. https://github.com/ibm-watson-data-lab/pixiedust/issues/683

2. Data scientists using Pixiedust can use custom color schemes to describe their data by generating accessibility-focused color palettes. https://github.com/ibm-watson-data-lab/pixiedust/issues/684

screenshot_overview

Authors:
Jeffrey Amari: @jaumb
Carla Aravena: @caravena555
Ye Ser Kim: @yk4at
Daniel Manning: @DanielManning
Alex Rice: @akr24

### CODE REVIEW

File: pixiedust/display/chart/renderers/templates/renderer.html [hunk 1]
```diff
@@ -30,6 +30,17 @@
                         data-default={{this.options.get(option.name,None) or option.metadata.default}}
                         style="margin: 0 0.6em;"></div>
                 </div>
+                {% if option.name == 'numbins' and this.get_numbins_options_pixieapp() %}
```
Comment (DTAIEB) [inline pixiedust/display/chart/renderers/templates/renderer.html]: I would prefer if we don't hardcode specific cases in the generic renderer.html. I suggest we define a new type of common options like pd_app that would generate a button pretty much the same way you're doing it now.

File: pixiedust/display/chart/renderers/templates/renderer.html [hunk 2]
```diff
@@ -30,6 +30,17 @@
...
+                    <div class="rendererOpt">
+                        <button pd_app="{{this.get_numbins_options_pixieapp()}}"
+                            class="btn btn-default no_loading_msg"
+                            pd_options="runInDialog=true"
+                            id="numbins_btn_{{prefix}}">
+                            <pd_event_handler pd_source="numbins_btn_{{prefix}}" pd_script="ShellAccess[eventInfo['entity']].on_ok()"></pd_event_handler>
```
Comment (DTAIEB) [inline pixiedust/display/chart/renderers/templates/renderer.html]: I believe the pd_event_handler is pretty generic, i.e. always calling the on_ok() method of the child PixieApp

**Comment by DanielManning** [comment]
I've created a generic button type that expects a name and description like the other input types, and in addition, expects the FQ name of the pd_app in its metadata.

File: pixiedust/display/chart/renderers/mapbox/mapBoxMapDisplay.py [hunk 3]
```diff
@@ -179,12 +181,31 @@ def renderMapView(self, mbtoken):
 
             # if there's a numeric value field and type is not 'simple', paint the data as a choropleth map
             if self.options.get("kind") and self.options.get("kind").find("simple") < 0 and len(valueFields) > 0:
+                # get value from the "Number of Bins" slider
+                numBins = 5 # default
```
Comment (DTAIEB) [inline pixiedust/display/chart/renderers/mapbox/mapBoxMapDisplay.py]: Nit pick: You may want to use this more concise syntax which makes it easier to read: 
numBins = self.options.get("numbins", 5)

**Comment by DanielManning** [comment]
Sounds good, did the same thing for the color fields.

**Comment by moekraft** [comment]
Great job team! You really exemplify the intent and value of the [IBM Developer JumpStart program](https://youtu.be/unY6txKTG1k)!

**Pull request merged**